### ISSUE TITLE
Ban `hmac/secrets.compare_digest()` on arbitrary strings

### ISSUE THREAD

**Issue by Viicos** [astral-sh/ruff#27309]
### Summary

[`hmac.compare_digest()`](https://docs.python.org/3/library/hmac.html#hmac.compare_digest) (re-expored under `secrets`) accepts strings or byte-like objects. However, only ASCII strings are allowed. This is documented, but not as a warning which makes it quite hard to catch. While the Python docs could be improved, it would be valuable to have a Ruff rule imo.

Suggested rule: ban using the function if a non-literal string is used (or a non-ASCII literal), and recommend encoding it first:

```python
hmac.compare_digest("a", "a")  # fine, known ASCII string literals
hmac.compare_digest("é", "é")  # bad, non ASCII

# This probably requires type-aware analysis:
def compare(a: str, b: str):
    return hmac.compare_digest(a, b)  # bad, `a` and `b` are not guaranteed to be ASCII
```

**Comment by jonathandung** [comment]
What about the private `_compare_digest` in `_operator`?
